p2p/metrics.go

```diff
@@ -79,6 +94,21 @@ func (c *peerPendingMetricsCache) AddPendingRecvBytes(msgType reflect.Type, addB
 	}
 }
 
+func (c *peerPendingMetricsCache) AddPendingMessageSendDelay(chID byte, delay time.Duration) {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	if entry, ok := c.perChannelChache[chID]; ok {
+		entry.count++
+		entry.sendDelay += delay
+	} else {
+		c.perChannelChache[chID] = &peerPendingDelayMetrics{
+			chID:      chID,
```

Comment:
Why do we need this field here (we have it as the map's key)?

Reply:
True, ehe.